plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java

```diff
@@ -229,28 +271,83 @@ private static boolean isLocationPresent(Storage storage)
         return storage.getOptionalLocation().isPresent() && isNotEmpty(storage.getLocation());
     }
 
-    /**
-     * The class enforces intentionally object identity semantics for the value holder,
-     * not value-based class semantics to correctly act as an invalidation guard in the
-     * cache.
-     */
     private static class ValueHolder
```

Comment:
I think that the name for this class is not fit anymore in the context of the new implementation.